File: java/src/main/java/com/dokmatiq/docgen/client/FontsClient.java
package com.dokmatiq.docgen.client;

import com.dokmatiq.docgen.internal.FileUtils;
import com.dokmatiq.docgen.internal.Transport;

import java.nio.file.Path;
import java.util.List;
import java.util.Map;

/** Client for font management endpoints. */
public class FontsClient {
    private final Transport transport;

    public FontsClient(Transport transport) {
        this.transport = transport;
    }

    /** Upload a font file (TTF/OTF). */
    public Map<?, ?> upload(Path file) {
        return transport.upload("/api/fonts", "file",
                FileUtils.readBytes(file), FileUtils.detectFilename(file), null, Map.class);
    }

    /** Upload a font from bytes. */
    public Map<?, ?> upload(byte[] data, String fileName) {
        return transport.upload("/api/fonts", "file", data, fileName, null, Map.class);
    }

    /** List all uploaded fonts. */
    public List<Map> list() {
        return transport.requestList("GET", "/api/fonts", Map.class);
    }

    /** Delete a font by name. */
    public void delete(String name) {
        transport.delete("/api/fonts/" + name);
    }
}

File: dotnet/src/DocGen/Clients/FontsClient.cs
using Dokmatiq.DocGen.Internal;

namespace Dokmatiq.DocGen.Clients;

/// <summary>
/// Client for font management endpoints.
/// </summary>
public sealed class FontsClient
{
    private readonly Transport _transport;

    internal FontsClient(Transport transport) => _transport = transport;

    /// <summary>Upload a font file (TTF/OTF).</summary>
    public Dictionary<string, object> Upload(string filePath)
        => _transport.Upload<Dictionary<string, object>>("/api/fonts", "file",
            FileUtils.ReadBytes(filePath), FileUtils.DetectFilename(filePath));

    /// <summary>Upload a font from bytes.</summary>
    public Dictionary<string, object> Upload(byte[] data, string fileName)
        => _transport.Upload<Dictionary<string, object>>("/api/fonts", "file", data, fileName);

    /// <summary>List all uploaded fonts.</summary>
    public List<Dictionary<string, object>> List()
        => _transport.RequestList<Dictionary<string, object>>(HttpMethod.Get, "/api/fonts");

    /// <summary>Delete a font by name.</summary>
    public void Delete(string name) => _transport.Delete($"/api/fonts/{name}");
}
